pkg/odo/cli/url/create.go

```diff
@@ -229,6 +276,27 @@ func (o *URLCreateOptions) Validate() (err error) {
 		} else if o.urlType == envinfo.INGRESS {
 			errorList = append(errorList, "host must be provided in order to create URLS of Ingress Kind")
 		}
+		if o.isDocker {
+			if len(o.exposure) > 0 {
+				errorList = append(errorList, "endpoint exposure is not supported in docker kind")
+			}
+			if len(o.protocol) > 0 {
+				errorList = append(errorList, "endpoint protocol is not supported in docker kind")
+			}
+			if len(o.path) > 0 {
+				errorList = append(errorList, "endpoint path is not supported in docker kind")
+			}
+			if o.secureURL {
+				errorList = append(errorList, "secure endpoint is not supported in docker kind")
+			}
```

Comment:
these flags are not present for the docker preference, no need to check

Reply:
updated